inc/classes/subscriber/Tools/class-cloudflare-subscriber.php

```diff
@@ -0,0 +1,888 @@
+				'message' => sprintf( __( 'Cloudflare cache purge error: %s', 'rocket' ), $cf_purge->get_error_message() ),
+			];
+		} else {
+			$cf_purge_result = [
+				'result'  => 'success',
+				'message' => __( 'Cloudflare cache successfully purged', 'rocket' ),
+			];
+		}
+
+		set_transient( get_current_user_id() . '_cloudflare_purge_result', $cf_purge_result );
+	}
+
+	/**
+	 * Purge CloudFlare cache
+	 *
+	 * @since 2.5
+	 */
+	public function do_purge_cloudflare() {
+		if ( ! isset( $_GET['_wpnonce'] ) || ! wp_verify_nonce( $_GET['_wpnonce'], 'rocket_purge_cloudflare' ) ) {
+			wp_nonce_ays( '' );
+		}
+
+		if ( ! current_user_can( 'rocket_purge_cloudflare_cache' ) ) {
+			return;
+		}
+
+		// Purge CloudFlare.
+		$cf_purge = $this->purge_cloudflare();
+
+		if ( is_wp_error( $cf_purge ) ) {
+			$cf_purge_result = [
+				'result'  => 'error',
+				// translators: %s = CloudFare API return message.
+				'message' => sprintf( __( '<strong>WP Rocket:</strong> %s', 'rocket' ), $cf_purge->get_error_message() ),
+			];
+		} else {
+			$cf_purge_result = [
+				'result'  => 'success',
+				'message' => __( '<strong>WP Rocket:</strong> Cloudflare cache successfully purged.', 'rocket' ),
+			];
+		}
+
+		set_transient( get_current_user_id() . '_cloudflare_purge_result', $cf_purge_result );
+
+		wp_safe_redirect( esc_url_raw( wp_get_referer() ) );
+		die();
+	}
+
+	/**
+	 * Set Real IP from CloudFlare
+	 *
+	 * @since 2.8.16 Uses CloudFlare API v4 to get CloudFlare IPs
+	 * @since 2.5.4
+	 * @source cloudflare.php - https://wordpress.org/plugins/cloudflare/
+	 */
+	public function set_real_ip() {
+		$is_cf = ( isset( $_SERVER['HTTP_CF_CONNECTING_IP'] ) ) ? true : false;
+
+		if ( ! $is_cf ) {
+			return;
+		}
+		// only run this logic if the REMOTE_ADDR is populated, to avoid causing notices in CLI mode.
+		if ( isset( $_SERVER['REMOTE_ADDR'] ) ) {
+			$cf_ips_values = $this->get_cloudflare_ips();
+
+			if ( strpos( $_SERVER['REMOTE_ADDR'], ':' ) === false ) {
+				$cf_ip_ranges = $cf_ips_values->result->ipv4_cidrs;
+
+				// IPV4: Update the REMOTE_ADDR value if the current REMOTE_ADDR value is in the specified range.
+				foreach ( $cf_ip_ranges as $range ) {
+					if ( rocket_ipv4_in_range( $_SERVER['REMOTE_ADDR'], $range ) ) {
+						if ( $_SERVER['HTTP_CF_CONNECTING_IP'] ) {
+							$_SERVER['REMOTE_ADDR'] = $_SERVER['HTTP_CF_CONNECTING_IP'];
+						}
+						break;
+					}
+				}
+			} else {
+				$cf_ip_ranges = $cf_ips_values->result->ipv6_cidrs;
+
+				$ipv6 = get_rocket_ipv6_full( $_SERVER['REMOTE_ADDR'] );
+				foreach ( $cf_ip_ranges as $range ) {
+					if ( rocket_ipv6_in_range( $ipv6, $range ) ) {
+						if ( $_SERVER['HTTP_CF_CONNECTING_IP'] ) {
+							$_SERVER['REMOTE_ADDR'] = $_SERVER['HTTP_CF_CONNECTING_IP'];
+						}
+						break;
+					}
+				}
+			}
+		}
+
+		// Let people know that the CF WP plugin is turned on.
+		if ( ! headers_sent() ) {
+			header( 'X-CF-Powered-By: WP Rocket ' . WP_ROCKET_VERSION );
+		}
```

Comment:
I'm questioning if we should keep that part, does it bring any value for anyone? Not sure the support team uses it @wp-media/supportrocket